clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -181,6 +215,15 @@ public synchronized void update(Cluster cluster, long now) {
     }
 
     /**
+     * If expiry is enabled, remove the unknown topic from the metadata refresh set to
+     * avoid further metadata requests for deleted topics which are not in use.
+     */
+    public synchronized void handleUnknownTopic(String topic) {
```

Comment:
Do we really need this method? It seems like we could pass the relevant information in the `update` method. Not yet sure which way is better, but I'd like to consider the option.